openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -32,59 +32,65 @@ function getJqueryElements(selector){
-                if (data.name == '' || data.name == '---') {
-                    return error('#id-errors', 'select-id', dataConfig['Please select an identifier.'])
-                }
-                const label = $('#select-id').find(`option[value=${data.name}]`).html();
-                if (data.value == '') {
-                    return error('#id-errors', 'id-value', dataConfig['You need to give a value to ID.'].replace(/ID/, label));
-                }
-                if (['ocaid'].includes(data.name) && /\s/g.test(data.value)) {
-                    return error('#id-errors', 'id-value', dataConfig['ID ids cannot contain whitespace.'].replace(/ID/, label));
-                }
-                if (data.name == 'isbn_10' && data.value.length != 10) {
-                    return error('#id-errors', 'id-value', dataConfig['ID must be exactly 10 characters [0-9] or X.'].replace(/ID/, label));
-                }
-                if (data.name == 'isbn_13' && data.value.replace(/-/g, '').length != 13) {
-                    return error('#id-errors', 'id-value', dataConfig['ID must be exactly 13 digits [0-9]. For example: 978-1-56619-909-4'].replace(/ID/, label));
-                }
-                $('id-errors').hide();
-                return true;
+            if (data.name == '') {
+                return error('#role-errors', '#role-name', dataConfig['You need to give this ROLE a name.'].replace(/ROLE/, data.role));
             }
-        });
-        $('#classifications').repeat({
-            vars: {prefix: 'edition--'},
-            validate: function (data) {
-                if (data.name == '' || data.name == '---') {
-                    return error('#classification-errors', '#select-classification', dataConfig['Please select a classification.']);
-                }
-                if (data.value == '') {
-                    const label = $('#select-classification').find(`option[value=${data.name}]`).html();
-                    return error('#classification-errors', '#classification-value', dataConfig['You need to give a value to CLASS.'].replace(/CLASS/, label));
-                }
-                $('#classification-errors').hide();
-                return true;
+            $('#role-errors').hide();
+            return true;
+        }
+    });
+}
+
+export function initIdentifierValidation() {
+    const dataConfig = JSON.parse(document.querySelector('#identifiers').dataset.config);
+    $('#identifiers').repeat({
+        vars: {prefix: 'edition--'},
+        validate: function (data) {
+            if (data.name == '' || data.name == '---') {
+                return error('#id-errors', 'select-id', dataConfig['Please select an identifier.'])
             }
-        });
+            const label = $('#select-id').find(`option[value=${data.name}]`).html();
+            if (data.value == '') {
+                return error('#id-errors', 'id-value', dataConfig['You need to give a value to ID.'].replace(/ID/, label));
```

Comment:
Let's use a more easy-to-remember easy-to-write  id for this.
Also rather than replacing an ID let's use $1 to avoid false positives and distinguish from plain text... consider a translation "You need to give a value to the IDENTIFIER ID"

e.g. `dataConfig.emptyStringError.replace(/$1/, label)`

Presumably we want to internationalize this later. I suggest just replace

Reply:
I would prefer to keep this PR scoped to moving the JS. Doing this change would require touching the translations and I'm not familiar with them or able to get different languages to work on my local machine. So you can make a separate issue but as Drini said these will likely be converted to Vue in the medium term so maybe the change can be saved until then.

Reply:
No need to touch translations. My concern is around the key in the object we pass to the client, and the confusion we add to the code by using the long key names

Here's what I would suggest:

https://gist.github.com/jdlrobson/d4d1cc997edb138015ae05989ffb3dfd